spec/services/user_signup_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'spec_helper'
+
+describe UserSignup do
+  describe ".perform" do
+    subject(:perform) do
+      described_class.perform(user: user,
+                              invitation_token: invitation_token,
+                              stripe_token: stripe_token)
+    end
+    let(:stripe_token) { "garble" }
+    let(:invitation_token) { nil }
+
+    context "with valid personal information" do
+      let(:user) do
+        Fabricate.build(:user,
+                        fullname: "Joe Doe",
+                        email: 'joe@doe.com')
+      end
+
+      before do
+        expect(StripeWrapper::Charge).to receive(:create).and_return(charge)
+      end
+
+      context "and valid credit card" do
+        let(:charge) { double(:charge, successful?: true) }
+
+        after { ActionMailer::Base.deliveries.clear }
+
+        it "creates new user" do
+          perform
```

Comment:
On 1 side this looks nice, - not many details. On the other side, your specs are documenting the code and refactoring some details above will make it more unclear itself. So, imagine you have bug or error at certain lines. You will follow to those lines and then you will have to most likely check out the lines that are way above.

When you have before blocks and lots of context, reading and understanding 1 spec will be way more difficult. Refactoring and drying specs out is not always a good option to do.